## Session III: RNA-Seq Part II

### Description

Session III will start with an interactive lesson in R, wherein we will cover advanced topics including the Tidyverse suite of data science packages. We will then use R to perform the differential gene expression (DGE) analysis and generate a list of differentially expressed (DE) genes using the Salmon abundance estimates from the MOV10 experiment, and the associated metadata file as input. In addition to performing this analysis, we will be talking about QC steps and statistics that go into the DGE analysis. The list(s) of DE genes will then be used for performing a functional interpretation of results.

**Day1:** The first day will start with more advanced concepts in R, including an introduction to the Tidyverse suite of data science packages. Then, using the R know-how, participants will begin the DGE analysis of the RNA-Seq dataset by using the abundance estimates output from Salmon and the experimental design file. Participants will begin to learn in-depth about each step in the DGE analysis workflow, including performing QC on the count data prior to the step for identifying DE genes. 

**Day2:** On the second day, the DGE analysis workflow will continue and DESeq2 will be employed to identify DE genes. In addition participants will employ various tools to graphically represent the results of this analysis. In the afternoon, participants will learn how to use AnnotationHub and two Gene Ontology-based functional enrichment analysis tools (clusterProfiler, gProfileR).   

### Lessons
[Click here for the schedule with links to the lessons.](schedule)

### Learning Objectives
* Demonstrate how to use the Tidyverse syntax
* Examine the quality of quantified data.
* Describe considerations for performing statistical analysis on RNA-Seq data.
* Employ DESeq2 to obtain a list of significantly different genes.
* Employ DESeq2 specific functions to transform and/or visualize data
* Perform functional analysis on gene lists with R-based tools.
