## Session IV: Isoform-level differential expression, automation, and other RNA-seq applications

### Description

Session IV will start with a discussion on the considerations for isoform level differential expression, followed by an interactive lesson on the analysis of isoform level differential expression of Salmon abundance estimates. Then, we will combine the different components of RNA-seq analysis to create our own automated workflow/pipeline using shell scripting, followed by a hands-on lesson on how to use HBC's bcbio-nextgen pipeline for RNA-seq analysis. Finally, we will explore other NGS technologies related to RNA-seq, with a focus on small RNA-seq and single cell RNA-seq.

**Day1:** We will start the first day discussing methods and considerations for performing an isoform-level differential expression analysis and use Sleuth on the output of Salmon to obtain lists of differentially expressed transcripts/genes. We will then learn how to automate the RNA-seq workflow for improved efficiency and how to use the [bcbio-nextgen RNA-seq](https://bcbio-nextgen.readthedocs.io/en/latest/index.html) pipeline, which accepts FASTQ data and experimental metadata as input, and returns abundance estimates as output. At the end of the day, we will explore other NGS applications related to RNA-seq, including small RNA-seq with Dr. Lorena Pantano. 

**Day2:** We will spend the day exploring single-cell RNA-seq, starting the day with an overview of single-cell RNA-seq and experimental considerations by Dr. Sarah Boswell and a discussion of analytical steps by Dr. Ayshwayra Subramanian. We will start the hands-on analysis of data after lunch, including QC analysis, clustering and marker identification. We will end the day with a discussion of single-cell analysis with the invited experts.    

### Lessons
[Click here for the schedule with links to the lessons.](schedule)
